dev_guide/builds.adoc

```diff
@@ -846,43 +850,48 @@ source:
+Streaming content in binary format from a local file system to the builder is called a `*binary type build*`.
+The corresponding value of `*BuildConfig.spec.source.type*` is `*Binary*` for such builds.
 
-A binary source potentially extracts content (if `*asFile*` is not set), in which case `*contextDir*`
-allows changing to a subdirectory within the content before the build executes.
-
-The source definition is part of the `*spec*` section in the `*BuildConfig*`:
+This source type is unique in that it is leveraged solely based on your use of the `oc start-build`.
 
+[NOTE]
 ====
-[source,yaml]
-----
-source:
-  type: "Binary"
-  binary: <1>
-    asFile: "webapp.war" <2>
-  contextDir: "app/dir" <3>
-  dockerfile: "FROM centos:7\nRUN yum install -y httpd" <4>
-----
-<1> The `*binary*` field specifies the details of the binary source.
-<2> The `*asFile*` field specifies the name of the file that will be created with
-the binary contents.
-<3> The `*contextDir*` field specifies a subdirectory within the extracted contents of a
-binary archive.
-<4> If the optional `*dockerfile*` field is provided, it should be a string
-containing an inline Dockerfile that potentially replaces one within the
-contents of the binary archive.
+Since binary type builds require content to be streamed from the local file system, it is not possible
+to automatically trigger a binary type build (e.g. via an image change trigger) because the binary files could
+not be provided. Similarly, you cannot launch binary type builds from the web console.
 ====
 
+To utilize binary builds, invoke `oc start-build` with one of these options:
+
+1) `--from-file`:  The contents of the file you specify will be sent as a binary stream to the builder.  Then
+the builder will store the data in a file with the same name at the top of the build context.
+
+2) `--from-dir` and `--from-repo`:  The contents will be archived and sent as a binary stream to the builder.
+Then the builder will extract the contents of the archive within the build context directory.
+
+In each of the above cases:
+
+1) If your `*BuildConfig*` already has a `*Binary*` source type defined, it will effectively be ignored and replaced by what the
+client sends.
+
+2) If your `*BuildConfig*` has a `*Git*` source type defined, it will be dynamically disabled, since `*Binary*` and `*Git*` are
+mutually exclusive, and the data in the binary stream provided to the builder will take precedence.
+
+When using `oc new-build --binary=true`, the command ensures that the restrictions associated with binary builds are enforced.
+The resulting `*BuildConfig*` will have a source type of `*Binary*`, meaning that the only valid way to run a build for this `*BuildConfig*`
+will be to use `oc start-build` with one of the `--from` options to provide the requisite binary data.
+
+The `*dockerfile*` and `*contextDir*` link:#source-code[Git source options] have special meaning with binary builds.
```

Comment:
they are just source options, not git source options (ie they are applicable even when you don't have a git source).  the section you linked to needs some reworking/retitling to clarify that, but we can do that in another PR.